velox/expression/SignatureBinder.cpp

```diff
@@ -69,8 +70,21 @@ bool SignatureBinder::tryBind(
   auto it = bindings_.find(typeSignature.baseType());
   if (it == bindings_.end()) {
     // concrete type
-    if (boost::algorithm::to_upper_copy(typeSignature.baseType()) !=
-        actualType->kindName()) {
+    auto typeName = boost::algorithm::to_upper_copy(typeSignature.baseType());
+    if (isDecimalName(typeName)) {
+      VELOX_USER_FAIL("Use 'DECIMAL' in the signature.");
+    }
```

Comment:
isDecimalName() sounds like typeName is DECIMAL and it is an error to use it. Is it possible to rename this function.

Reply:
I can add a comment here to clarify. But I don't see an easy name fix. We have `isDecimalKind` and `isDecimalName` in `Type.h`, which will be used everywhere. `DECIMAL` is only used in TypeSignature.